microceph/ceph/manager.go

```diff
@@ -27,3 +30,28 @@ func bootstrapMgr(hostname string, path string) error {
 func joinMgr(hostname string, path string) error {
 	return bootstrapMgr(hostname, path)
 }
+
+// Add Ceph Mgr service on host.
+func addMgr(hostname string) error {
+	pathConsts := common.GetPathConst()
+	pathFileMode := common.GetPathFileMode()
+
+	mgrDataPath := filepath.Join(pathConsts.DataPath, "mgr", fmt.Sprintf("ceph-%s", hostname))
+
+	err := os.MkdirAll(mgrDataPath, pathFileMode[pathConsts.DataPath])
+	if err != nil {
+		return fmt.Errorf("Failed to join manager: %w", err)
+	}
+
+	err = joinMgr(hostname, mgrDataPath)
+	if err != nil {
+		return fmt.Errorf("Failed to join manager: %w", err)
+	}
+
+	err = snapStart("mgr", true)
+	if err != nil {
+		return fmt.Errorf("Failed to start manager: %w", err)
+	}
+
+	return nil
+}
```

Comment:
This pattern of adding a service by creating a directory, joining and starting the snap seems to be repeated a couple of times. Would it be worthy to create a helper for this?